linkcheck/htmlutil/linkname.py

```diff
@@ -52,7 +52,7 @@ def image_name (txt):
 def href_name (txt):
     """Return the name part of the first <a href="">name</a> link in txt."""
     name = u""
-    endtag = a_end_search(txt)
+    endtag = a_end_search(urlutil.decode_for_unquote(txt))
     if not endtag:
         return name
     name = txt[:endtag.start()]
```

Comment:
Same problem as before: you're using Unicode code point positions to slice a byte string.  This will break with non-ASCII characters.

Reply:
I was going with:
```diff
@@ -19,7 +19,7 @@ Parse names of title tags and link types.
 """
 
 import re
-from .. import HtmlParser, strformat, url as urlutil
+from .. import HtmlParser, strformat
 
 
 imgtag_re = re.compile(r"(?i)\s+alt\s*=\s*"+\
@@ -31,7 +31,7 @@ def endtag_re (tag):
     """Return matcher for given end tag"""
     return re.compile(r"(?i)</%s\s*>" % tag)
 
-a_end_search = endtag_re("a").search
+a_end_search = endtag_re(b"a").search
 title_end_search = endtag_re("title").search
 
 
@@ -52,7 +52,7 @@ def image_name (txt):
 def href_name (txt):
     """Return the name part of the first <a href="">name</a> link in txt."""
     name = u""
-    endtag = a_end_search(urlutil.decode_for_unquote(txt))
+    endtag = a_end_search(txt)
     if not endtag:
         return name
     name = txt[:endtag.start()]
```

but then I found there is already a later commit which does:
```diff
def href_name (txt):
    """Return the name part of the first <a href="">name</a> link in txt."""
    name = u""
-   endtag = a_end_search(urlutil.decode_for_unquote(txt))
+   txt = urlutil.decode_for_unquote(txt)
+   endtag = a_end_search(txt)
    if not endtag:
        return name
    name = txt[:endtag.start()]
```


https://github.com/linkchecker/linkchecker/pull/40/commits/bc0e3f475a8a0b7b96d3f7b43094c1e7b02d4e01#diff-67af5e271aa8b2b09086214b45adef39

Any preferences?

Reply:
:+1: for decoding the entire text before processing

Reply:
OK. Changed to the latter.